### PR TITLE
Added support to PostgreSQL identity column

### PR DESCRIPTION
**Pull request by diegosardina**
- bug fix
- BC break? no

(Sorry for the new PR but I deleted the old branch and repository)

Since PostgreSQL 10, identity columns are recommended instead of serial types as stated in the documentation (and serial types are even discouraged [here](https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_serial)).

In the old PR `a.attidentity = ANY (ARRAY['a', 'd'])` and `pg_get_serial_sequence()` were used.  While the first only caused problem with older PostgreSQL, the latter has (known) issues with multiple schemas. The best way was to directly query pg_catalog (that is not something trival).

Tests are successful (here) both with serial types and identity columns. Does it make sense to keep both or we will update the current ones to use just the new identity column?

### CODE REVIEW

**Comment by milo** [comment]
I would stay with SERIAL support, PostgreSQL 9.6 has no identities and is still supported (well, to November 2021).

There is one bug in this PR, missing parenthesis around new OR operands. In that case, it marks autoincrement column event if not primary key. Sorry, there is no test for it.

Please, fix your commit (or I can do that, I'll leave you as author but signature will be missing). And I have prepeared test too.

File: src/Database/Drivers/PgSqlDriver.php [hunk 1]
```diff
@@ -135,13 +135,15 @@ public function getColumns(string $table): array
 				CASE WHEN a.atttypmod = -1 THEN NULL ELSE a.atttypmod -4 END AS size,
 				NOT (a.attnotnull OR t.typtype = 'd' AND t.typnotnull) AS nullable,
 				pg_catalog.pg_get_expr(ad.adbin, 'pg_catalog.pg_attrdef'::regclass)::varchar AS default,
-				coalesce(co.contype = 'p' AND strpos(pg_catalog.pg_get_expr(ad.adbin, ad.adrelid), 'nextval') = 1, FALSE) AS autoincrement,
+				coalesce(co.contype = 'p' AND seq.relname IS NOT NULL OR strpos(pg_catalog.pg_get_expr(ad.adbin, ad.adrelid), 'nextval') = 1, FALSE) AS autoincrement,
```
Comment (milo) [inline src/Database/Drivers/PgSqlDriver.php:138]: there must be parentheses `AND ( ... OR ... ) `

**Comment by milo** [comment]
One day, query may be siplified to use `pg_attribute.attidentity` column since PostgreSQL 10.

File: src/Database/Drivers/PgSqlDriver.php (hunk 1 above)
Comment (diegosardina) [reply src/Database/Drivers/PgSqlDriver.php:138]: Ops, sorry...

**Comment by diegosardina** [comment]
> One day, query may be siplified to use `pg_attribute.attidentity` column since PostgreSQL 10.

Yup, also the latter line may be simplified by using pg_get_serial_sequence() when they fix problems with multiple schemas.

Thanks for the review :-)

**Comment by milo** [comment]
Thank you. I squashed the commits and merged them as c46a1dececf8146d78367b3012e07ad80c89c93f

**Pull request closed without merging**